File: Phase01/src/InvertedIndexMaker.java
import java.util.HashMap;

public class InvertedIndexMaker {
    private final InvertedIndex invertedIndex;
    private final FileReader fileReader;

    public InvertedIndexMaker(FileReader fileReader) {
        this.invertedIndex = new InvertedIndex();
        this.fileReader = fileReader;
    }

    public void make() throws InputException {
        fileReader.readFiles();
        HashMap<String, String> docNameToContent = fileReader.getDocNameToContent();

        for (String docName : docNameToContent.keySet()) {
            invertedIndex.indexDocument(docNameToContent.get(docName), docName);
        }
    }

    public InvertedIndex getInvertedIndex() {
        return invertedIndex;
    }
}

File: Phase05/SampleLibrary/SampleLibrary/InvertedIndexMaker.cs
namespace SampleLibrary;

public class InvertedIndexMaker
{
    private readonly Dictionary<string, string> _documents;

    public InvertedIndexMaker(Dictionary<string, string> documents)
    {
        _documents = documents;
    }

    public InvertedIndex Make(InvertedIndex invertedIndex)
    {
        foreach (var doc in _documents)
            invertedIndex.indexDocument(doc.Value, doc.Key);
        return invertedIndex;
    }
}